### ISSUE TITLE
Import app_localizations fails after build_runner - Flutter 3.22

### ISSUE THREAD

**Issue by EliaTolin** [flutter/flutter#148333]
### Steps to reproduce

Using Flutter 3.22
1. `flutter pub run build_runner watch --delete-conflicting-outputs`

### Expected results

Build runner ends successfully without recurring problems to app_localization.

### Actual results

After launching build_runner the import of app_localization fails:

```
The target URI does not exist: 'package:flutter_gen/gen_l10n/app_localizations.dart'.
Try creating the file referenced by the URI, or try using a URI for a file that exists.
```

To solve this you need to do after build runner:

``` 
flutter gen-l10n
flutter pub get 
```

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter_gen/gen_l10n/app_localizations.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
            Text(AppLocalizations.of(context).hello),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.22.0, on macOS 14.4.1 23E224 darwin-arm64, locale en-IT)
    • Flutter version 3.22.0 on channel stable at /Users/eliatolin/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5dcb86f68f (5 days ago), 2024-05-09 07:39:20 -0500
    • Engine revision f6344b75dc
    • Dart version 3.4.0
    • DevTools version 2.34.3

[!] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/eliatolin/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.15.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2023.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.7+0-17.0.7b1000.6-10550314)

[✓] VS Code (version 1.89.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.88.0
```

</details>

**Comment by pascalfriedrich** [comment]
Same for our project as well, except that we aren't be able to use `flutter gen-l10n` because of:
```
[  +81 ms] Because l10n.yaml exists, the options defined there will be used instead.
           To use the command line arguments, delete the l10n.yaml file in the Flutter project.
```

I mean, we could use the command line arguments for this as a temporary workaround, but... meh. :D 

Btw: I opened the same issue on the build_runner repo, because I thought that's a build_runner issue related to flutter 3.22: https://github.com/dart-lang/build/issues/3693

**Comment by EliaTolin** [comment]
> try to add `flutter pub get` after `dart run build_runner build`. the whole part in my github actions like this below:
> 
> ```
>   - name: Generate build runner
>     run: |
>       dart run build_runner build --delete-conflicting-outputs
>       flutter pub get
> ```

Thank you for this workaround. 
But the problem persist when i develop with `build_runner watch `

**Comment by darshankawar** [comment]
Thanks for the report @EliaTolin 
I am unable to replicate the reported behavior using stable 3.22.0
Can you elaborate as what exactly is the behavior you are seeing as compared to 3.19 ?

**Comment by sarevd** [comment]
> try to add `flutter pub get` after `dart run build_runner build`. the whole part in my github actions like this below:
> 
> ```
>   - name: Generate build runner
>     run: |
>       dart run build_runner build --delete-conflicting-outputs
>       flutter pub get
> ```

Thanks for the workaround it works with Gitlab CI.

I had the issue only on CI (error • Target of URI doesn't exist: 'package:flutter_gen/gen_l10n/app_localizations.dart'), locally it was fine.

I also tried to modify your workaround like:
      dart run build_runner build --delete-conflicting-outputs
      flutter pub get
      dart run build_runner build --delete-conflicting-outputs
      
But it reproduced the same error again. My guess is that flutter pub get removes the flutter_gen/gen_l10n

@darshankawar maybe this info will be helpful

**Comment by EliaTolin** [comment]
> Thanks for the report @EliaTolin I am unable to replicate the reported behavior using stable 3.22.0 Can you elaborate as what exactly is the behavior you are seeing as compared to 3.19 ?

@darshankawar Thank you for your reply, yes of course.

Taking the code sample up and this files:



#### pubspec.yaml 

```
name: bug
description: "A new Flutter project."
publish_to: 'none' 
version: 1.0.0+1

environment:
  sdk: '>=3.4.0 <4.0.0'

dependencies:
  flutter:
    sdk: flutter

  flutter_localizations:
    sdk: flutter
  
  cupertino_icons: ^1.0.6
  intl: ^0.19.0

dev_dependencies:
  flutter_test:
    sdk: flutter

  flutter_lints: ^3.0.0
  build_runner: ^2.4.9

flutter:
  generate: true
  uses-material-design: true

```

#### l10n.yaml


```
arb-dir: lib/l10n
template-arb-file: app_en.arb
output-localization-file: app_localizations.dart
nullable-getter: false
```

#### lib/app_en.arb

`{"hello":"Hello World"}`


If we generate app_localizations throught run : 

`flutter gen-l10n`

and after this we want generate files throught run:

`flutter pub run build_runner watch --delete-conflicting-outputs`

after the import of app_localizations.dart fails:

`The target URI does not exist: 'package:flutter_gen/gen_l10n/app_localizations.dart'.
Try creating the file referenced by the URI, or try using a URI for a file that exists.`

It's like when do flutter clean but only for l10n files, other imports work perfectly.

I am available for further testing

**Comment by darshankawar** [comment]
Thanks for the updates. I was able to replicate the reported behavior with 3.22.0, but not in 3.16.9. 
Based on it, I'll label this as a regression.

<details>
<summary> stable, master flutter doctor -v </summary>

```
[!] Flutter (Channel stable, 3.22.0, on macOS 12.2.1 21D62 darwin-x64, locale
    en-GB)
    • Flutter version 3.22.0 on channel stable at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5dcb86f68f (5 days ago), 2024-05-09 07:39:20 -0500
    • Engine revision f6344b75dc
    • Dart version 3.4.0
    • DevTools version 2.34.3
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

[!] Flutter (Channel master, 3.22.0-30.0.pre.11, on macOS 12.2.1 21D62
    darwin-x64, locale en-GB)
    • Flutter version 3.22.0-30.0.pre.11 on channel master at
      /Users/dhs/documents/fluttersdk/flutter
    ! Warning: `flutter` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
      your current Flutter SDK checkout at
      /Users/dhs/documents/fluttersdk/flutter. Consider adding
      /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
    ! Warning: `dart` on your path resolves to
      /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
      current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
      Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
      of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c90e18c6f5 (2 hours ago), 2024-05-09 20:32:11 -0700
    • Engine revision c0917b14fc
    • Dart version 3.5.0 (build 3.5.0-138.0.dev)
    • DevTools version 2.36.0-dev.5
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/dhs/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for
      more details.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13C100
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 65.1.4
    • Dart plugin version 213.7228

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.29.0

[✓] Connected device (3 available)
    • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
      • iOS 15.3.1 19D52
    • macOS (desktop)           • macos                                    •
      darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)              • chrome                                   •
      web-javascript • Google Chrome 109.0.5414.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
      
[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (5 available)
    • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 98.0.4758.80

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.



```

</details>

**Comment by Jaew00Shin** [comment]
> Thanks for the updates. I was able to replicate the reported behavior with 3.22.0, but not in 3.16.9. Based on it, I'll label this as a regression.
> 
> stable, master flutter doctor -v
> ```
> [!] Flutter (Channel stable, 3.22.0, on macOS 12.2.1 21D62 darwin-x64, locale
>     en-GB)
>     • Flutter version 3.22.0 on channel stable at
>       /Users/dhs/documents/fluttersdk/flutter
>     ! Warning: `flutter` on your path resolves to
>       /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
>       your current Flutter SDK checkout at
>       /Users/dhs/documents/fluttersdk/flutter. Consider adding
>       /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
>     ! Warning: `dart` on your path resolves to
>       /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
>       current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
>       Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
>       of your path.
>     • Upstream repository https://github.com/flutter/flutter.git
>     • Framework revision 5dcb86f68f (5 days ago), 2024-05-09 07:39:20 -0500
>     • Engine revision f6344b75dc
>     • Dart version 3.4.0
>     • DevTools version 2.34.3
>     • If those were intentional, you can disregard the above warnings; however
>       it is recommended to use "git" directly to perform update checks and
>       upgrades.
> 
> [!] Xcode - develop for iOS and macOS (Xcode 12.3)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     ! Flutter recommends a minimum Xcode version of 13.
>       Download the latest version or update via the Mac App Store.
>     • CocoaPods version 1.11.2
> 
> [✓] Chrome - develop for the web
>     • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
> 
> [✓] VS Code (version 1.62.0)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     • Flutter extension version 3.21.0
> 
> [✓] Connected device (5 available)
>     • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
>     • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
>       ios            • iOS 14.4.1 18D61
>     • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
>       ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
>     • macOS (desktop)            • macos                                    •
>       darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
>     • Chrome (web)               • chrome                                   •
>       web-javascript • Google Chrome 98.0.4758.80
> 
> [✓] HTTP Host Availability
>     • All required HTTP hosts are available
> 
> ! Doctor found issues in 1 category.
> 
> [!] Flutter (Channel master, 3.22.0-30.0.pre.11, on macOS 12.2.1 21D62
>     darwin-x64, locale en-GB)
>     • Flutter version 3.22.0-30.0.pre.11 on channel master at
>       /Users/dhs/documents/fluttersdk/flutter
>     ! Warning: `flutter` on your path resolves to
>       /Users/dhs/Documents/Fluttersdk/flutter/bin/flutter, which is not inside
>       your current Flutter SDK checkout at
>       /Users/dhs/documents/fluttersdk/flutter. Consider adding
>       /Users/dhs/documents/fluttersdk/flutter/bin to the front of your path.
>     ! Warning: `dart` on your path resolves to
>       /Users/dhs/Documents/Fluttersdk/flutter/bin/dart, which is not inside your
>       current Flutter SDK checkout at /Users/dhs/documents/fluttersdk/flutter.
>       Consider adding /Users/dhs/documents/fluttersdk/flutter/bin to the front
>       of your path.
>     • Upstream repository https://github.com/flutter/flutter.git
>     • Framework revision c90e18c6f5 (2 hours ago), 2024-05-09 20:32:11 -0700
>     • Engine revision c0917b14fc
>     • Dart version 3.5.0 (build 3.5.0-138.0.dev)
>     • DevTools version 2.36.0-dev.5
>     • If those were intentional, you can disregard the above warnings; however
>       it is recommended to use "git" directly to perform update checks and
>       upgrades.
> 
> [!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
>     • Android SDK at /Users/dhs/Library/Android/sdk
>     ✗ cmdline-tools component is missing
>       Run `path/to/sdkmanager --install "cmdline-tools;latest"`
>       See https://developer.android.com/studio/command-line for more details.
>     ✗ Android license status unknown.
>       Run `flutter doctor --android-licenses` to accept the SDK licenses.
>       See https://flutter.dev/docs/get-started/install/macos#android-setup for
>       more details.
> 
> [✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Build 13C100
>     • CocoaPods version 1.11.2
> 
> [✓] Chrome - develop for the web
>     • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
> 
> [✓] IntelliJ IDEA Ultimate Edition (version 2021.3.2)
>     • IntelliJ at /Applications/IntelliJ IDEA.app
>     • Flutter plugin version 65.1.4
>     • Dart plugin version 213.7228
> 
> [✓] VS Code (version 1.62.0)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     • Flutter extension version 3.29.0
> 
> [✓] Connected device (3 available)
>     • Darshan's iphone (mobile) • 21150b119064aecc249dfcfe05e259197461ce23 • ios
>       • iOS 15.3.1 19D52
>     • macOS (desktop)           • macos                                    •
>       darwin-x64     • macOS 12.2.1 21D62 darwin-x64
>     • Chrome (web)              • chrome                                   •
>       web-javascript • Google Chrome 109.0.5414.119
> 
> [✓] Network resources
>     • All expected network resources are available.
> 
> ! Doctor found issues in 1 category.
>       
> [!] Xcode - develop for iOS and macOS (Xcode 12.3)
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     ! Flutter recommends a minimum Xcode version of 13.
>       Download the latest version or update via the Mac App Store.
>     • CocoaPods version 1.11.2
> 
> [✓] Chrome - develop for the web
>     • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
> 
> [✓] VS Code (version 1.62.0)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     • Flutter extension version 3.21.0
> 
> [✓] Connected device (5 available)
>     • SM G975F (mobile)       • RZ8M802WY0X • android-arm64   • Android 11 (API 30)
>     • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
>       ios            • iOS 14.4.1 18D61
>     • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
>       ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
>     • macOS (desktop)            • macos                                    •
>       darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
>     • Chrome (web)               • chrome                                   •
>       web-javascript • Google Chrome 98.0.4758.80
> 
> [✓] HTTP Host Availability
>     • All required HTTP hosts are available
> 
> ! Doctor found issues in 1 category.
> ```

When do you expect this issue to be resolved?

**Comment by Jaew00Shin** [comment]
@darshankawar When do you expect this issue to be resolved?

**Comment by rddewan** [comment]
same error here

**Comment by Senne021** [comment]
Same error

**Comment by plotic** [comment]
Getting the error too
[✓] Flutter (Channel stable, 3.22.0)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web

**Comment by krille-chan** [comment]
Same error here experiencing on Flutter (Channel stable, 3.22.0) with:

- Github Actions (latest Ubuntu)
- macOS 14.4.1 23E224 darwin-arm64
- Also on my Ubuntu 24.04 machine (Flutter installed manually and **not** as a snap)

**Comment by florinleu** [comment]
Same error here
- Flutter (Channel stable, 3.22.0)
- macOS 14.4.1

**Comment by AhmedBassiouny** [comment]
Same error here

Flutter (Channel stable, 3.22.0)
macOS 14.4.1

**Comment by stuartmorgan** [comment]
Anyone considering commenting here: please read https://github.com/flutter/flutter/wiki/Issue-hygiene#do-not-add-me-too-or-same-or-is-there-an-update-comments-to-bugs first.

**Comment by bstolinski** [comment]
Workaround
```
arb-dir: lib/l10n
template-arb-file: app_en.arb
synthetic-package: false
output-dir: lib/l10n/gen_l10n
output-localization-file: app_localizations.dart
nullable-getter: false
```

and run `flutter gen-l10n`

**Comment by jxstxn1** [comment]
Running `flutter pub deps` triggers the same behaviour.

**Comment by rao21** [comment]
Same error any update on fix ?

**Comment by darrenjacquet** [comment]
I am also having the same issue and cannot run dart run build_runner watch and have the app_localizations.dart available. It seem the only way to get this to work is to run dart run build_runner and then flutter pub get each time. This is a total pain. Please fix and patch soon.

**Comment by FabrizioBilleciUNICT** [comment]
Faced the same issue. So absurd to have that on prod channel.

**Comment by dberchtenbreiter-beurer** [comment]
> Faced the same issue. So absurd to have that on prod channel.

I think it works with 3.22.1

**Comment by Dohmanlechx** [comment]
The 3.22.1 update resolved my issue in this regard. Thank you Flutter team, for your quick and hard work.

Friendly reminder to everyone that there is a command: `flutter downgrade` which downgrades your Flutter to the previous version. The process is literally instantaneous (unlike `flutter upgrade`).

**Comment by andrewkolos** [comment]
Closing as fixed by https://github.com/dart-lang/sdk/issues/55758, which was included in the 3.22.1 hotfix release.
